src/utils/tests/nviHelpers.test.ts

```diff
@@ -0,0 +1,157 @@
+    updatedRegistration.entityDescription.publicationDate!.year = '2001';
+
+    const result = await willResetNviStatuses(persistedRegistration, updatedRegistration);
+    expect(result).toBe(true);
+  });
+
+  test('Returns false when date is changed within same year', async () => {
+    let persistedRegistration = structuredClone(nviRegistration);
+    persistedRegistration.entityDescription.publicationDate!.day = '1';
+    persistedRegistration.entityDescription.publicationDate!.month = '1';
+    persistedRegistration.entityDescription.publicationDate!.year = '2000';
+
+    let updatedRegistration = structuredClone(nviRegistration);
+    updatedRegistration.entityDescription.publicationDate!.day = '2';
+    updatedRegistration.entityDescription.publicationDate!.month = '2';
+    updatedRegistration.entityDescription.publicationDate!.year = '2000';
+
+    const result = await willResetNviStatuses(persistedRegistration, updatedRegistration);
+    expect(result).toBe(false);
+  });
+
+  test('Returns true when category is changed', async () => {
+    let updatedRegistration = structuredClone(nviRegistration);
+    updatedRegistration.entityDescription.reference!.publicationInstance.type = JournalType.AcademicLiteratureReview;
+
+    const result = await willResetNviStatuses(nviRegistration, updatedRegistration);
+    expect(result).toBe(true);
+  });
+
+  test('Returns true when journal or publisher is changed', async () => {
+    const channelId1 = 'https://api.com/channel/1';
+    const channelId2 = 'https://api.com/channel/2';
+
+    let persistedRegistration = structuredClone(nviRegistration);
+    persistedRegistration.entityDescription.reference!.publicationContext.id = channelId1;
+
+    let updatedRegistration = structuredClone(persistedRegistration);
+    updatedRegistration.entityDescription.reference!.publicationContext.id = channelId2;
+
+    const result = await willResetNviStatuses(persistedRegistration, updatedRegistration);
+    expect(result).toBe(true);
+  });
+
+  test('Returns true when series is changed', async () => {
+    const seriesId1 = 'https://api.com/channel/1';
+    const seriesId2 = 'https://api.com/channel/2';
+
+    let persistedRegistration = structuredClone(nviRegistration) as BookRegistration;
+    persistedRegistration.entityDescription.reference!.publicationContext = {
+      type: PublicationType.Book,
+      isbnList: [],
+      seriesNumber: '',
+      series: {
+        type: PublicationChannelType.Series,
+        id: seriesId1,
+      },
+    };
+
+    let updatedRegistration = structuredClone(persistedRegistration) as BookRegistration;
+    updatedRegistration.entityDescription.reference!.publicationContext.series!.id = seriesId2;
+
+    const result = await willResetNviStatuses(persistedRegistration, updatedRegistration);
+    expect(result).toBe(true);
+  });
+
+  test('Returns true when a new institution is added as affiliation', async () => {
```

Comment:
Må mocke requests for å få til dette, siden det avhenger av institusjonstreet
https://vitest.dev/guide/mocking.html#requests